### PR TITLE
[TopBar] Animatable theming

### PR DESCRIPTION
**Pull request by kaelig**
### WHY are these changes introduced?

References https://github.com/Shopify/destinations/issues/863

### WHAT is this pull request doing?

Introduces transitions on elements that change color/background-color when the theme of the top bar is switched after the page has loaded. See issue above for the full context.

Note: this doesn't follow the design specification to the letter, as some tweaks were made by @AlexBlaise and I directly in the browser in order to reach a result that wouldn't trigger disproportionate amounts of engineering work.

This PR is safe to revert when we have the ability in `Shopify/web` to determine these conditions server-side rather than client-side: **user has access to a single Plus store in an organization, and the Global nav beta is enabled in this store**. When this can be computed server-side, we'll instantly render the appropriate theme, rather than animating it in.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/TopBar/components/SearchDismissOverlay/SearchDismissOverlay.scss [hunk 1]
```diff
@@ -2,7 +2,7 @@
 @import '../../variables';
 
 $entry-iterations: 1;
-$backdrop-color: rgba(33, 43, 54, 0.4);
+$backdrop-color: rgba(color('ink'), 0.4);
```
Comment (kyledurand) [inline src/components/TopBar/components/SearchDismissOverlay/SearchDismissOverlay.scss:5]: Can this use a custom property with a color(‘ink’) fallback?

File: src/components/TopBar/components/SearchField/SearchField.scss [hunk 2]
```diff
@@ -98,6 +98,7 @@ $stacking-order: (
 
   &::placeholder {
     color: var(--p-text, var(--top-bar-color));
+    transition: (duration() - 33ms) color easing() 33ms;
```
Comment (kyledurand) [inline src/components/TopBar/components/SearchField/SearchField.scss:101]: Super nit but 33ms can be a sass var.

File: src/components/TopBar/components/Menu/Menu.scss [hunk 3]
```diff
@@ -2,7 +2,10 @@
 
 $activator-variables: (
   min-width: rem(36px),
-  transition: background-color 0.1s,
+  transition: (
+    duration(fast) background-color,
```
Comment (kyledurand) [inline src/components/TopBar/components/Menu/Menu.scss:6]: Should background color have easing as well?

File: src/components/TopBar/components/SearchField/SearchField.scss (hunk 2 above)
Comment (kaelig) [reply src/components/TopBar/components/SearchField/SearchField.scss:101]: I was thinking that as well, but wasn't sure where to put it so it'd be available across multiple partials. Where would it make sense for this Sass var to be stored?

File: src/components/TopBar/components/SearchDismissOverlay/SearchDismissOverlay.scss (hunk 1 above)
Comment (kaelig) [reply src/components/TopBar/components/SearchDismissOverlay/SearchDismissOverlay.scss:5]: Theming of the search experience is out of scope for this piece of work, would it be okay to revisit this at a later time, if it becomes needed?

File: src/components/TopBar/components/Menu/Menu.scss (hunk 3 above)
Comment (kaelig) [reply src/components/TopBar/components/Menu/Menu.scss]: Actually, let me remove the changes I made to this map. The color transition is applied to children, not on the top level activator element.

File: src/components/TopBar/components/SearchField/SearchField.scss (hunk 2 above)
Comment (BPScott) [reply src/components/TopBar/components/SearchField/SearchField.scss:101]: If we need a place to share sass variables across subcomponents we occasionally put an `_variables.scss` file in the component folder. For TopBar we've already got  https://github.com/Shopify/polaris-react/blob/master/src/components/TopBar/_variables.scss

**Pull request merged**